test/ng/directive/ngModelSpec.js

```diff
@@ -455,6 +455,42 @@ describe('ngModel', function() {
     });
 
 
+    describe('$setModelValue', function() {
+
+      it('should set the value to $modelValue', function() {
+        ctrl.$setModelValue(10);
+        expect(ctrl.$modelValue).toBe(10);
+      });
+
+      it('should update the value on the scope', inject(function($compile) {
```

Comment:
Actually looking at this again I don't like that we can set the `$modelValue` to a value that is different to the scope.

For your use case @realityking you should always call this method as `ngModel.$setModelValue(ngModel.$modelValue)` and providing a different value should be documented as not supported.
This would remove the need for this test and the associated behaviour.